include/solvers/mpm_base.tcc

```diff
@@ -904,6 +911,78 @@ void mpm::MPMBase<Tdim>::nodal_frictional_constraints(
   }
 }
 
+// Nodal pore pressure constraints (Coupled solid-fluid formulation)
+template <unsigned Tdim>
+void mpm::MPMBase<Tdim>::nodal_pore_pressure_constraints(
```

Comment:
As we write pore_pressure in state_vars, maybe we can combine the two constraints function?